tests/unit-tests/tools/dds/rtpsrelaylib/Utility.cpp

```diff
@@ -0,0 +1,228 @@
+#ifdef OPENDDS_HAS_CXX11
+
+#include <dds/rtpsrelaylib/Utility.h>
+
+#include <gtest/gtest.h>
+
+using namespace RtpsRelay;
+
+TEST(tools_dds_rtpsrelaylib_Utility, guid_to_string)
+{
+  OpenDDS::DCPS::GUID_t guid = OpenDDS::DCPS::GUID_UNKNOWN;
+  EXPECT_EQ(guid_to_string(guid), "00000000.00000000.00000000.00000000");
+}
+
+TEST(tools_dds_rtpsrelaylib_Utility, AddrPort_default_ctor)
+{
+  AddrPort ap;
+  EXPECT_EQ(ap.addr, ACE_INET_Addr());
+  EXPECT_EQ(ap.port, SPDP);
+}
+
+TEST(tools_dds_rtpsrelaylib_Utility, AddrPort_ctor)
+{
+  ACE_INET_Addr addr("127.0.0.1");
+  AddrPort ap(addr, DATA);
+  EXPECT_EQ(ap.addr, addr);
+  EXPECT_EQ(ap.port, DATA);
+}
+
+TEST(tools_dds_rtpsrelaylib_Utility, AddrPort_equal)
+{
+  ACE_INET_Addr addr("127.0.0.1:1234");
+  AddrPort ap1(addr, DATA);
+  AddrPort ap2(addr, DATA);
+  EXPECT_EQ(ap1, ap2);
+}
+
+TEST(tools_dds_rtpsrelaylib_Utility, AddrPort_not_equal)
+{
+  ACE_INET_Addr addr1("192.168.1.1:1234");
+  ACE_INET_Addr addr2("192.168.1.2:1234");
+  AddrPort ap1(addr1, DATA);
+  AddrPort ap2(addr1, SPDP);
+  AddrPort ap3(addr2, DATA);
+  AddrPort ap4(addr2, SPDP);
+  EXPECT_NE(ap1, ap2);
+  EXPECT_NE(ap1, ap3);
+  EXPECT_NE(ap1, ap4);
+  EXPECT_NE(ap2, ap3);
+  EXPECT_NE(ap2, ap4);
+  EXPECT_NE(ap3, ap4);
+}
+
+TEST(tools_dds_rtpsrelaylib_Utility, AddrPort_less_than)
+{
+  ACE_INET_Addr addr1("192.168.1.1:1234");
+  ACE_INET_Addr addr2("192.168.1.2:1234");
+  AddrPort ap1(addr1, SPDP);
+  AddrPort ap2(addr1, DATA);
+  AddrPort ap3(addr2, SPDP);
+  AddrPort ap4(addr2, DATA);
+  EXPECT_LT(ap1, ap2);
+  EXPECT_LT(ap2, ap3);
+  EXPECT_LT(ap3, ap4);
+}
+
+TEST(tools_dds_rtpsrelaylib_Utility, GuidAddr_ctor)
+{
+  OpenDDS::DCPS::GUID_t guid;
+  ACE_INET_Addr addr("127.0.0.1");
+  AddrPort ap(addr, SPDP);
+  GuidAddr ga(guid, ap);
+  EXPECT_EQ(ga.guid, guid);
+  EXPECT_EQ(ga.address, ap);
+}
+
+TEST(tools_dds_rtpsrelaylib_Utility, GuidAddr_equal)
+{
+  OpenDDS::DCPS::GUID_t guid;
+  ACE_INET_Addr addr("127.0.0.1:1234");
+  AddrPort ap(addr, SPDP);
+  GuidAddr ga1(guid, ap);
+  GuidAddr ga2(guid, ap);
+  EXPECT_EQ(ga1, ga2);
+}
+
+TEST(tools_dds_rtpsrelaylib_Utility, GuidAddr_not_equal)
+{
+  OpenDDS::DCPS::GUID_t guid1;
```

Comment:
```suggestion
  OpenDDS::DCPS::GUID_t guid1 = OpenDDS::DCPS::GUID_UNKNOWN;
```

To get rid of the warnings?